server/index.js

```diff
@@ -23,103 +23,128 @@ const open = require('open');
 const portfinder = require('portfinder');
 const crypto = require("crypto");
 const fse = require('fs-extra');
+const { createHttpTerminator } = require('http-terminator');
 
+const DEFAULT_PORT = '9000';
 
-async function start(portPreferred) {
-    let port = process.env.ASSET_COMPUTE_DEV_PORT || '9000';
-    if (!isNaN(portPreferred)) {
-        port = portPreferred;
+class DevtoolServer {
+
+    async run (portPreferred) {
+        this.port = process.env.ASSET_COMPUTE_DEV_PORT || DEFAULT_PORT;
+        if (!isNaN(portPreferred)) {
+            this.port = portPreferred;
+        }
+    
+        const portRangeServer = {
+            port: this.port, // minimum port
+            stopPort: 9999 // maximum port
+        };
+        this.port = await findOpenPort(portRangeServer);
```

Comment:
Careful about typing for `this.port` - see comment about. Are both Number types and strings supported? Otherwise you may need to ensure conversion.

Reply:
[`portfinder.getPortPromise()`](https://www.npmjs.com/package/portfinder) ensures to return a valid open port